src/magfie/bdivfree.f90

```diff
@@ -181,6 +181,51 @@ subroutine vector_potentials(nr_in,np_in,nz_in,ntor_in,      &
 102 format(1000e15.7)
 end subroutine vector_potentials
 
+
+! Spline a single toroidal Fourier mode of the RMP coil field, as needed for MEPHIT.
+! Note the explicit-shape input arrays are not checked for shape.
+! Also, the truncation of data outside the limiting convex is not performed.
+subroutine vector_potential_single_mode(ntor_in, nR_in, nZ_in, Rmin_in, Rmax_in, Zmin_in, Zmax_in, Bn_R, Bn_Z)
+  use iso_fortran_env, only: dp => real64
+  use bdivfree_mod, only: nR, nZ, Rmin, Zmin, ntor, icp, ipoint, hR, hZ, Rpoi, Zpoi, AZnRe, AZnIm, ARnRe, ARnIm
```

Comment:
Are all these stored variables needed later? Or can one read some quantities from the size of arrays to reduce the argument list and make everything less side-effecty?

Reply:
They are used by the spline evaluation routines later and are saved just like in `vector_potentials`. To properly access array sizes, you would have to put these routines into a module (and give them proper names while you're at it). That's not the purpose of this pull request, but one of the distractions you would warn me about. But when we have time™ later, I can change it and you can test it.